### DISCUSSION TITLE
Что я делаю не так?

### DISCUSSION THREAD

**Discussion by WilhelmWas** [maxsite/cms#460] [General]
Я в issues ответа не дождался...
Быть может... сложный. я вопрос задал...
А может,.. не туда... я постучался???
А может непонятно написал???

Вопрос такой, мне нужно взять из БАЗЫ...
Да через функцию, из опций взять язык,
Любой другой... определяет сразу! -
А с русским... ну никак! ни тык ни мык...

Быть может больше ЗДЕСЬ народу ходит
Дискуссий нет Друзья, я не хочу!
Но mso_get_ОПШИН "за нос водит"
И может тут ответ я получу???

# Что я делаю не так?
$sys_lang = mso_get_option('admin_lang', 'general', []);

pr('$sys_lang = ' . $sys_lang);  // Работает при любом другом языке, кроме русского :-(

# На gettype() отзывается string, но строка не проверяется
# ни с NULL, ни с '',.. как победить? 
pr(gettype(mso_get_option('admin_lang', 'general', [])));  // получим string
# Благодарю за помощь

**Comment by maxsite** [comment]
Лето же...

Код верный. Язык по умолчанию русский — это пустая строка. Остальные языки  
ua || Українська 
ro || Romanian
en || English
de || Deutsch

**Comment by WilhelmWas** [comment]
День вчера (ЛЕТНИЙ, но не в Герм..) наверно был "хороший" (для меня :-))
И такой вот... не работал код!!! (СЕРЬЁЗНО!!!)
Комп напевно був мій несерйозний...
Или в нём полазил "старый кот" (шучу... Лето же!)

**$sys_lang** = mso_get_option('admin_lang', 'general', []);

if ($sys_lang == '') $sys_lang = 'ru'; 

pr('$sys_lang = ' . $sys_lang);  // Любой, кроме рус. подставлялся!

ТАКИМ вот кодом, я не мог получить русский!!! - потому и написал... что с пустой строкой не сравнивается
Код ведь, абсолютно верный и в стиле CMS!
Сейчас... вот... работает!!! Благодарю за содействие.

P.S. Лето,.. отдыхайте, веселитесь...
Пейте пiво, ешьте шашлыки...
с CMS работать не ленитесь...
И кропите верные строки...
Хорошего Дня, ну и т.д.

**Comment by maxsite** [comment]
Лучше так:

```
$sys_lang = mso_get_option('admin_lang', 'general', 'ru');

if ($sys_lang === '') $sys_lang = 'ru';
```

**Reply by WilhelmWas** [reply]
Верно! Так надёжней
